### PR TITLE
Updates supporting PubSub+ broker 9.8 or later

### PR DESCRIPTION
**Pull request by bczoma**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Updates supporting PubSub+ broker 9.8 or later

### CODE REVIEW

**Comment by bczoma** [comment]
Looking to address the failed checks

**Comment by bczoma** [comment]
Cannot reasonably address more best practices fails without adding complex logic or failing elsewhere.

USGov test is failing on `'DockerExtension', and type handler version '1.0' could not be found in the extension repository`.
not sure where to go from here, may need to change AzQuickStarts-USGov to unsupported.
@bmoore-msft - Brian, would you know the answer to this?

File: solace-message-router/azuredeploy.json [hunk 1]
```diff
@@ -513,14 +596,14 @@
...
-              "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('nicName'), copyindex()))]"
+              "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(parameters('vmAndLoadbalancerExposure'),variables('nicName'), copyindex()))]"
             }
           ]
         }
       }
     },
     {
-      "apiVersion": "2019-12-01",
+      "apiVersion": "2019-07-01",
```
Comment (bmoore-msft) [inline solace-message-router/azuredeploy.json:606]: why are we reverting to an older version?

File: solace-message-router/README.md [hunk 2]
```diff
@@ -62,21 +69,24 @@ You need to fill in the following fields (* marks the mandatory ones):
 | **BASICS**                 |  |
 | Subscription*              | Provide your subscription to use. |
 | Resource Group*            | A new group, or an existing group that will be available from the pull-down menu once "Use existing" is selected. |
-| Location*                  | Specify a location for the resource group where metadata about the resources is stored. Select the region most suitable to you. |
```
Comment (bmoore-msft) [inline solace-message-router/README.md:65]: location is still a param in the template - is this removal intentional?

File: solace-message-router/README.md [hunk 3]
```diff
@@ -62,21 +69,24 @@ You need to fill in the following fields (* marks the mandatory ones):
...
 | **SETTINGS**               |  |
+| Region*                  | Specify an Azure region for the deployment. Note that not every resource is available in every region. |
```
Comment (bmoore-msft) [inline solace-message-router/README.md:73]: I don't see region...

File: solace-message-router/azuredeploy.json [hunk 4]
```diff
@@ -42,6 +42,7 @@
       },
       "allowedValues": [
         "East US",
+        "US Gov Virginia",
```
Comment (bmoore-msft) [inline solace-message-router/azuredeploy.json:45]: I think you can remove this - the vm extension is not published to USGov... also update metadata.json to exclude USGov support (and remove the button from the readme (if present) - keep the badges)

Also remove the defaultValue as that tends to overload regions

File: solace-message-router/azuredeploy.parameters.us.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (bmoore-msft) [inline solace-message-router/azuredeploy.parameters.us.json:1]: can remove this file

File: solace-message-router/nestedtemplates/workspace-enabled-shared-resources.json [hunk 6]
```diff
@@ -16,6 +16,7 @@
       },
       "allowedValues": [
         "East US",
+        "US Gov Virginia",
```
Comment (bmoore-msft) [inline solace-message-router/nestedtemplates/workspace-enabled-shared-resources.json:19]: remove this

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: solace-message-router/azuredeploy.json (hunk 1 above)
Comment (bczoma) [reply solace-message-router/azuredeploy.json:606]: The best practice check script didn't recognize `"apiVersion": "2019-12-01"` so changed it to the closest it listed. Sounds like not an issue so reverting.

File: solace-message-router/README.md (hunk 2 above)
Comment (bczoma) [reply solace-message-router/README.md:65]: Thanks for catching but intentional: This has removed a duplicate: `Location` is already listed further down with a default.

File: solace-message-router/README.md (hunk 3 above)
Comment (bczoma) [reply solace-message-router/README.md:73]: `Region` seems to be added automatically to the template, along with `Subscription` and `Resource Group`. Visible on the Azure console.

File: solace-message-router/azuredeploy.json (hunk 4 above)
Comment (bczoma) [reply solace-message-router/azuredeploy.json:45]: Thanks for advising, will remove.

File: solace-message-router/azuredeploy.parameters.us.json (hunk 5 above)
Comment (bczoma) [reply solace-message-router/azuredeploy.parameters.us.json:1]: done

File: solace-message-router/nestedtemplates/workspace-enabled-shared-resources.json (hunk 6 above)
Comment (bczoma) [reply solace-message-router/nestedtemplates/workspace-enabled-shared-resources.json:19]: done

**Comment by bczoma** [comment]
@bmoore-msft , thanks for all your comments, results look better now to me. Let me know please if there is anything else blocking merge.
